conseil-common/src/test/scala/tech/cryptonomic/conseil/common/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,116 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.concurrent.ExecutionContext
+
+import cats.effect._
+import org.http4s._
+import org.http4s.client.Client
+import fs2.Stream
+import io.circe.generic.auto._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.circe.CirceEntityEncoder._
+import org.scalatest.{Matchers, WordSpec}
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+class RpcClientTest extends WordSpec with Matchers {
+
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+  case class Block(hash: String)
+  case class Params(height: Int)
+
+  "Rpc Client" should {
+      "return case class for valid json" in {
+        val json = """[{
+        |  "result": { "hash": "abc" },
+        |  "id": "requestId"
+        |}]""".stripMargin
+        val request = Stream(RpcRequest("2.0", "getBlock", Params(1), "requestId"))
+        val response = Response[IO](
+          Status.Ok,
+          body = Stream(json).through(fs2.text.utf8Encode)
+        )
+        val client = new RpcClient[IO]("https://api-endpoint.com", 1, httpClient(response))
```

Comment:
How do you see if we wrap this block as a single function that takes the expected response json payload?

It would unclutter the test code from the details of how the http4s client works internally, which are not relevant to the test logic per-se.